### ISSUE TITLE
Just some suggestions

### ISSUE THREAD

**Issue by iamsoNewBee** [kotmatross28729/EnviroMine-continuation#13]
1,Radiation and Corrupted Broadcaster destory the sanity.
2,Cold or warm water tend to trun into naomal water( like foods rotting? )
3,Use snow ball and glass bottle only to creat cold water bottle( really convenient for survive mode )
4,Freezer generates freeze water, Esky generates cold water.

How do u think? Some maybe hard to achieve. I know little about modding and I just like to suggest XD.

**Comment by kotmatross28729** [comment]
Well, in general I agree, however:

* 1.1) Corrupted Broadcaster, since it is a block, you can change the "ambient sanity" for it in the config, the suggestion is to add it by  default, I suppose?  

* 1.2)  Radiation - I was already planning to do something similar with temperature, but it didn't work out, because for some reason, if for example I add 2 degrees to the temperature with rad = 200, then 2 degrees will be added every second, and the value will increase 36-38-40... I haven't figured out how to implement it correctly yet (I can, of course, use "=", but it literally breaks, uh, let's say everything)
The same with black lung / mesothelioma, I wanted the maximum air quality to decrease in the later stages of the disease, because of which you could spend less time in the mines and suffocate faster, this caused the same problem as with radiation

* 2.) I'm not sure that cooling for water + rotting will be good for server performance, because 2 counters at the same time... in any case, I can make it as a configuration option

Otherwise, I completely agree

**Comment by kotmatross28729** [comment]
5.1) Quite strange, even baked foods contain some amount of water, the salt content is more important here. 

5.2) I don’t think that the temperature increase will be strong enough to take it into account (maybe a few hundredths of a degree, which is significantly small, because only tenths of degrees matter)

6.) As far as I know, this already exists, although I’ll double-check later

**Comment by iamsoNewBee** [comment]
For the 6: They are actually still rotting

**Comment by kotmatross28729** [comment]
**3)** Added*
**6)** Added*
**7)** Added*
**8)** Added* (+15.0 instead of +5.0)

"*" - in dev (not yet released)

---

I can’t say about everything else, I’ve added what I thought was necessary for now

I’ll add a little to the suggestions :

* Compatibility with Pam's HarvestCraft (food that restores hydration) - planned (soon) 
  - Maybe even compatibility with the JAFFA (HarvestCraft addon)?

* Compatibility with Hbm's NTM drinks (also hydration) - in progress

What do you think?

**Comment by iamsoNewBee** [comment]
I fully agree. And speaking of hbm, there are lots of canned food. I think it would be good if their shelf life is longer(or even not expire) by default.

**Comment by kotmatross28729** [comment]
Considering how long they lay in abandoned places, I think the shelf life should be unlimited, yes

**Comment by kotmatross28729** [comment]
(btw, I added you to the credits, your suggestions are pretty good!)

**Comment by iamsoNewBee** [comment]
Thanks a lot! I have used this mod for years. It's my pleasure to do some contributions for such an old mod. Thank you for maintaining this and accept my thoughts and turn them into reality😊👍👍👍

**Comment by iamsoNewBee** [comment]
Add to the suggestion 3. As there is  frozen water  now, maybe this is better:
Snowball + Empty Bottle = Frozen Water Bottle
Snowball + Water Bottle = Cold Water Bottle

**Comment by kotmatross28729** [comment]
As for heating/cooling water, a new system is being developed:

<details>

<summary>For some reason the link doesn't work, I'll copy it (comment 2654825904 on devlog)</summary>

#### In general, now recipes fully use autogen (that is, I don’t need to write 150000 `GameRegistry.addShapelessRecipe`, and remember all the combinations)

Let me briefly explain how it works:

### Each water type has 4 parameters: 1) `heatIndex`, 2) `isRadioactive`, 3) `isDirty`, 4) `isSalty`

---

## 1) `heatIndex` - temperature index.

For normal temperature water (`RADIOACTIVE`, `DIRTY`, `SALTY`, `CLEAN`) it is 0

For all 4 `COLD` it is -1

For `RADIOACTIVE_FROSTY` and `FROSTY` it is -2

For all 4 `WARM` it is 1

For `RADIOACTIVE_HOT` and `HOT` it is 2

`heatIndex` is used in generating recipes related to temperature changes

When you heat water, this index increases by 1, and when you cool it, it decreases by 1

As an example, let's take the water type `CLEAN`, it has a `heatIndex` of 0 , if we put the water type `CLEAN` in the furnace, the resulting water will be water with `heatIndex` 0+1=1

Since `CLEAN` doesn't have additional traits, we will skip them

Then the system finds water in accordance with the new parameters, that is, with `heatIndex` 1 and without additional parameters

This type of water is `CLEAN_WARM`.

Therefore, when heating, `CLEAN` -> `CLEAN_WARM`

---

## 2) `isRadioactive` - indicates that the water is radioactive

The strongest trait, applies to absolutely all types of water, cannot be removed

If you drink water with this trait, you will get a little radiation (if Hbm's NTM is installed)

Only found in NTM crater biomes

---

## 3) `isDirty` - indicates that the water is dirty

Applies to all types of water except `FROSTY` and `HOT`, can be removed

For removal by heat, the original water must have `heatIndex` >= 0, otherwise it will remain dirty (e.g. `DIRTY_COLD`, when heated gives `DIRTY`)

For removal by cold, the original water must have `heatIndex` < 0, otherwise it will remain dirty

---


## 4) `isSalty` - indicates that the water is salty

Exactly the same as `isDirty`:

Applies to all types of water except `FROSTY` and `HOT`, can be removed

To remove with heat, the original water must have `heatIndex` >= 0, otherwise it will remain salty (e.g. `SALTY_COLD`, when heated gives `SALTY`)

To remove with cold, the original water must have `heatIndex` < 0, otherwise it will remain salty

---

#### Well, this is what it looks like (with new types of water):

![Image](https://github.com/user-attachments/assets/a9be4809-dee1-4ef0-958d-2c61cff40979)


</details>

So 
> Snowball + Water Bottle = Cold Water Bottle.

(almost) already there.

---

About

> Snowball + Empty Bottle = Frozen Water Bottle.

So, you just put a snowball in an empty bottle and get frosty water? In the current (new) system, you put a snowball in regular water to get cold water, and then put a snowball in cold water to get frosty water. So this water is designed as the one that is obtained in 2 stages (and requires water).

Although, still, yeah, I agree that snow can have another use besides cooling.

Maybe `snowball + empty bottle = dirty water`?

It will hardly be cold, because the snowball will melt (otherwise you'd just get a snow in a bottle), and it will hardly be clean, because if you immediately got clean water from snow, it would be quite easy to survive in cold biomes (since snowballs are very easy to get in large quantities) and, well, I think it will be like this

What do you think?

**Comment by iamsoNewBee** [comment]
`snowball + cold dirty water = frosty water`
I think here should be frosty dirty water?

**Comment by kotmatross28729** [comment]
> 3) isDirty - indicates that the water is dirty
Applies to all types of water except FROSTY and HOT, can be removed

**Comment by iamsoNewBee** [comment]
It is little strange why dirty cold water could clean itself by a snowball (which is dirty too)

**Comment by kotmatross28729** [comment]
Fair. 

Then:

`snowball + empty bottle = clear water`?

I think this is acceptable

**Comment by iamsoNewBee** [comment]
Add to the suggestion4, maybe we can use the lower chamber of the fridge to produce cold water (or frosty water?) (more efficiently?).

**Comment by kotmatross28729** [comment]
MrCrayfish's Furniture Mod?

**Comment by iamsoNewBee** [comment]
Yes.

**Comment by kotmatross28729** [comment]
Yeah, it's possible.

I think frosty water is more suitable

**Comment by iamsoNewBee** [comment]
9. Something for the sanity:

-Cats and dogs restore sanity (I dont know if it is possible?).
-When the player is in a "sitting posture" (like in a mine car, on a horse or sitting on a mod chair), restore sanity (slightly slower than sleeping).
- I suddenly forget what the third point is = (

**Comment by kotmatross28729** [comment]
1) It's already possible, in the mob config (Look in issues, there is a guide. Not sure if it can be made by default, I'll take a look)
2) It's possible
3) yes

But it's unlikely to happen soon, first I need to finish this **damn water/buckets**, and gas rework, then I can finish everything else

**Comment by iamsoNewBee** [comment]
Jusr following your own plan! Just giving you some inspiration XD

**Comment by iamsoNewBee** [comment]
10. Campfire (from the campfire backprot Mod) could clean water in the cauldron. (Like what the fire does. This method for purifying water is usually ignored because we have to always make a fire before it if we dont have hell rocks at the early stage. Too troublesome. The campfire could be a good replacement and give improtance to the cauldron.)

**Comment by kotmatross28729** [comment]
But this is already in the config by default:

https://github.com/kotmatross28729/EnviroMine-continuation/blob/5d1ef99a38a72a848188db9bc5db232cad702f70/src/main/java/enviromine/core/EM_Settings.java#L466

**Comment by iamsoNewBee** [comment]
Didnt even notice that =/

**Comment by iamsoNewBee** [comment]
11, Compat with growcraft waterbag? Dont know the new water system, but in the old version, the bag always gets clean water.

**Comment by kotmatross28729** [comment]
I even forgot about the existence of enviromine waterbag 💀

(regarding Growthcraft, why not, will be added a little later)
